src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -552,17 +552,8 @@ public BigInteger(ulong value)
         }
 
         public BigInteger(Single value)
+            : this((Double)value)
         {
-            if (Single.IsInfinity(value))
-                throw new OverflowException(SR.Overflow_BigIntInfinity);
-            if (Single.IsNaN(value))
-                throw new OverflowException(SR.Overflow_NotANumber);
```

Comment:
> Since there is no arithmetic expression as mentioned by @JonHanna the compiler cannot "optimize" something

Well, the arithmetic expression could be passed to the ctor by the caller.  Do we know for certain such optimizations might not kick in across inlining?